File: 코딩테스트 입문/ㄷ/등수 매기기.java
import java.util.*;

class Solution {
    public int[] solution(int[][] score) {
        int n = score.length;
        int[] ranks = new int[n];
        List<double[]> avgList = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            double average = (score[i][0] + score[i][1]) / 2.0;
            avgList.add(new double[]{average, i});
        }

        avgList.sort((a, b) -> Double.compare(b[0], a[0]));

        int rank = 1;
        for (int i = 0; i < n; i++) {
            if (i == 0 || avgList.get(i)[0] != avgList.get(i - 1)[0]) {
                rank = i + 1;
            }
            ranks[(int) avgList.get(i)[1]] = rank;
        }

        return ranks;
    }
}

File: 코딩테스트 입문/ㄷ/등수 매기기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<vector<int>> score) {
    int n = score.size();
    vector<pair<double, int>> avg(n);
    vector<int> ranks(n);

    for (int i = 0; i < n; i++) {
        double average = (score[i][0] + score[i][1]) / 2.0;
        avg[i] = {average, i};
    }

    sort(avg.begin(), avg.end(), greater<pair<double, int>>());

    int rank = 1;
    for (int i = 0; i < n; i++) {
        if (i == 0 || avg[i].first != avg[i - 1].first) {
            rank = i + 1;
        }
        ranks[avg[i].second] = rank;
    }

    return ranks;
}